services/dynamic-sidecar/tests/unit/test_api_containers.py

```diff
@@ -411,56 +423,79 @@ def _expected_error_string(status_code: int) -> dict[str, Any]:
 async def test_outputs_watcher_disabling(
     test_client: TestClient,
     mock_event_filter_enqueue: AsyncMock,
+    caplog_info_debug: LogCaptureFixture,
 ):
     assert isinstance(test_client.application, FastAPI)
     outputs_context: OutputsContext = test_client.application.state.outputs_context
     outputs_manager: OutputsManager = test_client.application.state.outputs_manager
     outputs_manager.task_monitor_interval_s = WAIT_FOR_OUTPUTS_WATCHER / 10
-    WAIT_FOR_EVENTS = outputs_manager.task_monitor_interval_s * 10
+    WAIT_PORT_KEY_PROPAGATION = outputs_manager.task_monitor_interval_s * 10
+    CALLS_RECEIVED_BY_EVENT_FILTER = 3
 
-    async def _create_file_in_random_dir_in_inputs() -> int:
+    async def _create_port_key_events() -> int:
         random_subdir = f"{uuid4()}"
 
         await outputs_context.set_file_type_port_keys([random_subdir])
-        await asyncio.sleep(WAIT_FOR_EVENTS)
-
+        await asyncio.sleep(WAIT_PORT_KEY_PROPAGATION)
+
+        # NOTE: normally below code bloc would generate
+        # CALLS_RECEIVED_BY_EVENT_FILTER == 3 events
+        # NOTE: sometimes the same event can be generated twice, see below
+        # <FileCreatedEvent: event_type=created, src_path='/tmp/.../file_0be4b1ae-ec65-44d5-b733-bf65822660cc', is_directory=False>
+        # <FileCreatedEvent: event_type=created, src_path='/tmp/.../file_0be4b1ae-ec65-44d5-b733-bf65822660cc', is_directory=False>
+        # <FileModifiedEvent: event_type=modified, src_path='/tmp/.../file_0be4b1ae-ec65-44d5-b733-bf65822660cc', is_directory=False>
+        # <FileClosedEvent: event_type=closed, src_path='/tmp/.../file_0be4b1ae-ec65-44d5-b733-bf65822660cc', is_directory=False>
         dir_name = outputs_context.outputs_path / random_subdir
-        dir_name.mkdir()
-
+        await mkdir(dir_name)
         async with aiofiles.open(dir_name / f"file_{uuid4()}", "w") as f:
             await f.write("ok")
 
-        dir_count = len(
-            [
-                1
-                for x in outputs_context.outputs_path.glob("*")
-                if not f"{x}".endswith(HIDDEN_FILE_NAME)
-            ]
+        async for attempt in AsyncRetrying(**_TENACITY_RETRY_PARAMS):
```

Comment:
I *did not mean this in my previous comment* but rather retry if the "original" assert check you had fails

Here you have added a new assert check on the event completion (which is quite code intrusive. See my comment above).

Reply:
let's look at it together tomorrow

Reply:
Pair reviewed, test was redesigned to run as fast as possible.